contracts/stabilizer/PSMRouter.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity ^0.8.4;
+
+import "./IPSMRouter.sol";
+import "./PegStabilityModule.sol";
+import "../Constants.sol";
+
+contract PSMRouter is IPSMRouter {
+
+    IPegStabilityModule public immutable psm;
+
+    constructor(IPegStabilityModule _psm) {
+        psm = _psm;
+        /// allow the PSM to spend all of our WETH
+        /// this contract should never have eth, but in case someone self destructs and sends us some eth
+        /// to try and grief us, that will not matter as we do not reference the contract balance
+        IERC20(address(Constants.WETH)).approve(address(_psm), type(uint256).max);
+    }
+
+    modifier ensure(uint256 deadline) {
+        require(deadline >= block.timestamp, "PSMRouter: order expired");
+        _;
+    }
+
+    /// @notice fallback function so that users can just send this contract eth and receive Fei in return
+    /// this function takes an address and minAmountOut as params from the calldata
+    fallback(bytes calldata data) external payable returns (bytes memory) {
+        (address to, uint256 minAmountOut) = abi.decode(data, (address, uint256));
+
+        _depositWethAndMint(to, minAmountOut);
```

Comment:
Don't like use of calldata here. to should be default msg.sender and minAmountOut should be default msg.value